src/main/java/com/jcabi/http/request/BaseRequest.java

```diff
@@ -181,6 +187,26 @@ public BaseRequest(final Wire wre, final String uri,
         final Iterable<Map.Entry<String, String>> headers,
         final String method, final byte[] body,
         final int cnct, final int rdd) {
+        this(wre, uri, headers, method, body, cnct, rdd, null);
```

Comment:
@joris-lambrechts Can we avoid `null` here by passing `SSLContext.getDefault()`?

Reply:
@andreoss `SSLContext.getDefault()` throws a checked `java.security.NoSuchAlgorithmException`. I could use Lomboks `lombok.SneakyThrows` to overcome this if allowed with a `@SneakyThrows(NoSuchAlgorithmException.class)`

Reply:
It's not possible when calling a sibling constructor. I tried to default it in the constructor with all the parameters but that causes test failures breaking the default behavior. I'll leave it as-is with the extra null check.

Reply:
@joris-lambrechts You can extract this to a private static method, just like in `TrustedWire`

Reply:
Is there a need to have this as the default behavior? Not all communication uses https, e.g. internal services might still use non-secure connections.